### PR TITLE
Fix/transaction functions

### PR DESCRIPTION
**Pull request by ropats16**
Fixes Transaction functions for node operations:

- Fixes Bundlr Version
- Updates Bundlr instance import in transaction function
- Removal of `File` type as `data` input param from `createTransaction` function

The following utility has been tested:
- `createTransaction`
- `signTransaction`
- `postTransaction`
- `createAndPostWBundlr`

These work with both `node18` and `node16`. For running the scripts in `node16` you need to pass some experimental flags as follows which has been confirmed by the Arweave Team as well:

`node --experimental-fetch --no-warnings file-name`

Here is a repo for testing scripts: https://github.com/ropats16/arweavekit-examples-test
This must be linked with `ArweaveKit` via `yarn link` for testing.

However, the Bundlr fix work only in the `node` environment due the the way it is being imported. This import has been implemented after consultation with the Bundlr team. They are aware of the errors that persist and some incompatibility with `arweave-js` but there is no workaround at the moment. Hence, all our transaction functions work only in the node environment at the moment and as long as we continue to support Bundlr in the same file there is no way to test the other functions in web apps.

A potential solution is separating Bundlr functions and implementing them individually for `node` and `web`.

**Update 2:**

- All the transaction functions work in a node environment
- Arweave based transactions work in web apps
- Bundlr transaction functions trigger with a window.ethereum object but due to some underlying errors they always fallback to an arweave web transaction for now

This can all be tested with https://github.com/ropats16/arweavekit-examples-test via `yarn link` or packing the PR with `yarn pack`

**Update 3:**

Local testing fixed for `wallet` and `arweave` based functions. Tests must be run individually as `yarn test <filename>.spec.ts`

### CODE REVIEW

**Comment by ropats16** [comment]
Latest update to the PR fixes the following:

- All the transaction functions work in a node environment
- Arweave based transactions work in web apps
- Bundlr transaction functions trigger with a window.ethereum object but due to some underlying errors they always fallback to an arweave web transaction for now

This can all be tested with https://github.com/ropats16/arweavekit-examples-test via `yarn link` or packing the PR with `yarn pack`

**Review by 7i7o — APPROVED** [review]
- All node testscripts work both on Node 18 and Node 16 👍 
- All Arweave functions work on web apps (in the Vite test project), bundlr falls back to arweave (also run in both 16 & 18 envs for the web app and arweavekit just in case) 👍 
- All arweavekit transaction tests work 👍

Ready to merge

**Review by t8 — APPROVED** [review]
LGTM. Nice work @ropats16!

File: package.json [hunk 1]
```diff
@@ -47,11 +47,13 @@
     "typescript": "^4.9.5"
   },
   "dependencies": {
-    "@bundlr-network/client": "^0.11.9",
+    "@bundlr-network/client": "0.11.9",
+    "@metamask/providers": "^11.1.0",
     "arweave": "^1.13.5",
     "arweave-mnemonic-keys": "^0.0.9",
+    "ethers": "^6.6.7",
```
[the remark this answers is not in the document]
Comment (ropats16) [reply package.json:54]: Yes, this was added to add support for WebBundlr

File: src/lib/graphql.ts [hunk 2]
```diff
@@ -133,3 +133,160 @@ function initArweave(gateway: string) {
 
   return Arweave.init(config);
 }
+
+export const ARWEAVE_GATEWAYS = [
+  'arweave.net',
+  'arweave.dev',
+  'g8way.io',
+  'arweave-search.goldsky.com',
```
[the remark this answers is not in the document]
Comment (ropats16) [reply src/lib/graphql.ts:141]: @kranthicodes will be better able to respond to this

[the remark this answers is not in the document]
Comment (kranthicodes) [reply src/lib/graphql.ts:141]: @t8 Goldsky GQL endpoint `arweave-search.goldsky.com/graphql` has the same schema as arweave.net. I'm not too sure about ar.io.

**Pull request merged**